docs/static/azure-module.asciidoc

```diff
@@ -0,0 +1,451 @@
+reads from the Event Hub. 
+
+
+[id="plugins-{type}s-{plugin}-max_batch_size"]
+===== `max_batch_size`
+
+* Value type is <<number,number>>
+* Default value is `125`
+
+Maximum number of events retrieved and processed together. A checkpoint is
+created after each batch. Increasing this value may help with performance, but
+requires more memory. 
+
+
+[id="plugins-{type}s-{plugin}-storage_connection"]
+===== `storage_connection`
+* Value type is <<string,string>>
+* No default value 
+
+Connection string for blob account storage. Blob account storage persists the
+offsets between restarts, and ensures that multiple instances of Logstash
+process different partitions. 
+When this value is set, restarts resume where processing left off.
+When this value is not set, the `initial_position` value is used on every restart.
+
+We strongly recommend that you define this value for production environments.
+
+
+[id="plugins-{type}s-{plugin}-storage_container"]
+===== `storage_container`
+* Value type is <<string,string>>
+* Defaults to the Event Hub name if not defined
+
+Name of the storage container used to persist offsets and allow multiple instances of {ls}
+to work together. 
+ 
+To avoid overwriting offsets, you can use different storage containers. This is
+particularly important if you are monitoring two Event Hubs with the same name.
+You can use the advanced configuration model to configure different storage
+containers. 
+
+
+[id="plugins-{type}s-{plugin}-threads"]
+===== `threads`
+* Value type is <<number,number>>
+* Minimum value is `2`
+* Default value is `4`
+  
+Total number of threads used to process events. The value you set here applies
+to all Event Hubs. Even with advanced configuration, this value is a global
+setting, and can't be set per event hub. 
+
+
+include::shared-module-options.asciidoc[]
+    
+[[run-azure]]
+==== Start the module 
+
+. Be sure that the `logstash.yml` file is <<configuring-azure,configured correctly>>. 
+. Run this command from the Logstash directory:
+
+["source","shell",subs="attributes"]
+-----
+bin/logstash
+-----
+
+[[exploring-data-azure]]
+==== Explore your data
+When the Logstash Azure module starts receiving events, you can begin using the
+packaged Kibana dashboards to explore and visualize your data. 
+
+To explore your data with Kibana:
+
+. Open a browser to http://localhost:5601[http://localhost:5601] (username:
+  "elastic"; password: "{pwd}")
+. Click *Dashboard*.
+. Select the dashboard you want to see.
+
+
+==== Azure module schema
+
+This module reads data from the Azure Event Hub and adds some additional structure to the data for Activity Logs and SQL Diagnostics. The original data is always preserved and any data added or parsed will be namespaced under 'azure'. For example, 'azure.subscription' may have been parsed from a longer more complex URN.
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Name |Description|Notes
```

Comment:
Tables can render strangely in our docs right now if there are long words (dotted names are often that problem) that result in text overrunning the navigation widget. Generally we try to avoid using complex tables (with more than two columns). At this point, I'd just check the output to see if there are overruns and use a different format strategy (maybe definition lists) if there are issues.

Reply:
Looks good on local build.  Will verify again.